compiler/rustc_codegen_llvm/src/coverageinfo/mod.rs

```diff
@@ -93,6 +97,9 @@ impl<'ll, 'tcx> CodegenCx<'ll, 'tcx> {
 
 impl<'tcx> CoverageInfoBuilderMethods<'tcx> for Builder<'_, '_, 'tcx> {
     fn init_coverage(&mut self, instance: Instance<'tcx>) {
+        if !self.coverage_context().is_some_and(|ctx| !ctx.is_initialized_for(&instance)) {
```

Comment:
This should be equivalent to `is_none_or(|ctx| ctx.is_initialized_for(&instance))`. However `is_none_or` has not been stabilized yet.